test/beacon/Voucher.test.ts

```diff
@@ -626,29 +626,349 @@ describe('Voucher', function () {
                     .to.be.equal(voucherRlcBalancePreClaim + taskPrice);
                 // Requester balance should stay unchanged.
                 expect(requesterRlcBalancePostClaim).to.be.equal(requesterRlcBalancePreClaim);
-                // Sponsored amount not should decrease
+                // Sponsored amount should stay unchanged
                 expect(await voucher.getSponsoredAmount(dealId)).to.be.equal(dealSponsoredAmount);
             }
         });
 
-        describe('Should claim task when deal is partially sponsored', async () => {});
+        describe('Should claim task when deal is partially sponsored', async () => {
+            it('Classic', async () => await runTest(voucherMatchOrders, claim));
+            it('Boost', async () => await runTest(voucherMatchOrdersBoost, claimBoost));
+
+            async function runTest(matchOrdersBoostOrClassic: any, claimBoostOrClassic: any) {
+                await addEligibleAssets([app, dataset]); // workerpool not eligible.
+                const dealNonSponsoredAmount = workerpoolPrice * volume;
+                const taskNonSponsoredAmount = dealNonSponsoredAmount / volume;
+                // Deposit non-sponsored amount for requester and approve voucher.
+                await iexecPocoInstance
+                    .transfer(requester, dealNonSponsoredAmount)
+                    .then((tx) => tx.wait());
+                await iexecPocoInstance
+                    .connect(requester)
+                    .approve(voucherAddress, dealNonSponsoredAmount)
+                    .then((tx) => tx.wait());
+                // Match orders
+                await matchOrdersBoostOrClassic();
+                const {
+                    voucherCreditBalance: voucherCreditBalancePreClaim,
+                    voucherRlcBalance: voucherRlcBalancePreClaim,
+                    requesterRlcBalance: requesterRlcBalancePreClaim,
+                } = await getVoucherAndRequesterBalances();
+                // The voucher should've partially sponsored the deal.
+                const dealSponsoredAmount = await voucher.getSponsoredAmount(dealId);
+                const taskSponsoredAmount = dealSponsoredAmount / volume;
+                expect(dealSponsoredAmount).to.be.equal(dealPrice - dealNonSponsoredAmount);
+                expect(taskSponsoredAmount).to.be.equal(taskPrice - taskNonSponsoredAmount);
+
+                // Claim
+                await expect(claimBoostOrClassic())
+                    .to.emit(voucherHub, 'VoucherRefunded')
+                    .withArgs(voucherAddress, taskSponsoredAmount)
+                    .to.emit(voucher, 'TaskClaimedWithVoucher')
+                    .withArgs(taskId);
```

Comment:
Maybe `.withArgs(taskId);` is redundant in these other tests since we already check that arg in the first test

Reply:
I agree 
[Remove redundant verifications](https://github.com/iExecBlockchainComputing/iexec-voucher-contracts/pull/21/commits/3e93fe96f6d6e01f4251070eaa49844c6f9c8ca8)